### ISSUE TITLE
Add GF profile METADATA check for designer string similarity 

### ISSUE THREAD

**Issue by davelab6** [googlefonts/fontbakery#3317]
## Observed behaviour

https://fonts.google.com/?query=jik has 3 families

* `Sunflower` by `JIKJISOFT`
* `Song Myung` by JIKJI`
* `Gaegu` by `JIKJI SOFT`

These should probably all have the same designer credit. 

## Expected behaviour

Add a fontbakery METADATA check that searches for existing designer strings and WARNs when similar ones are detected. I guess there are some simple python modules already published that can return string similarity, but perhaps a very simple one would be to count how many times each ordinal character is shared with existing strings, so the first character will likely always be shared with many others, but the 8th should very rarely be shared, and in the above case, a new designer `JIKJITYPE` would be shared by at least 3 others for the first 5 characters, so that is a strong match... 

## Resources and exact process needed to replicate

This might be another check that requires the entire GF git repo as a dependency, or a duplicate list of designer strings extracted from the repo into a fontbakery profile side file and updated from time to time, like the MSFT 4-char foundry code strings...

**Comment by felipesanches** [comment]
yeah, indeed the need for access to the entire repo (or some cached file derived form it) is what complicates this a bit. But I agree that it would be a nice check to have.

**Comment by simoncozens** [comment]
It looks like we're wanting to check the GF repo itself, rather than a font family; I think this needs a new `googlefonts_repo` profile or similar, and a new condition to pull down and cache the repo.

**Comment by felipesanches** [comment]
Not a new profile, but a new scope for checking, still in the same profile.
By "scope" I mean that we have checks that work on these scopes nowadays:

* **font**: a single font path
* **fonts**: multiple font files
* **ttFont**: a single `ttlib.TTFont` object
* **ttFonts**: multiple font objects
* **font_metadata**: a single font entry from the METADATA.pb file
* **family_metadata**: the family protobuf parsed from a METADATA.pb file
* **description**: the DESCRIPTION.html file
* etc

All of those are implemented by a FB condition. I think we could have a new condition to facilitate working with the GF repo contents as well. But this should not mean the creation of a new profile, because **profiles are defined by vendor needs instead of kinds of checks.**

**Comment by felipesanches** [comment]
All of Font Bakery's `"new check proposal"` issues are being either closed or migrated to the [Fontspector issue tracker](https://github.com/fonttools/fontspector/issues), as there's no intention to implement new checks on the Font Bakery codebase anymore.

If you believe this proposal is still valid, please post a message here asking for it to be migrated. Otherwise, on **July 20, 2025** (30 days from now) it will be closed due to inactivity.
